docs/source/_ext/probnum-config-options.py

```diff
@@ -0,0 +1,82 @@
+from docutils import nodes
+from docutils.statemachine import StringList
+from sphinx.util.docutils import SphinxDirective, switch_source_input
+
+import probnum as pn
+
+
+class ProbNumConfigOptions(SphinxDirective):
+    def run(self):
+        table = nodes.table(
+            "",
+            classes=[
+                "longtable",
+                "colwidths-auto",
+                # "colwidths-given",
+            ],
+        )
+
+        group = nodes.tgroup("", cols=3)
+        table.append(group)
+        group.append(nodes.colspec("", colwidth=35))
+        group.append(nodes.colspec("", colwidth=15))
+        group.append(nodes.colspec("", colwidth=50))
+
+        thead = nodes.thead("")
+        group.append(thead)
+
+        headrow = nodes.row("")
+        thead.append(headrow)
+
+        headrow.append(nodes.entry("", nodes.paragraph(text="Config Option")))
+        headrow.append(nodes.entry("", nodes.paragraph(text="Default Value")))
+        headrow.append(nodes.entry("", nodes.paragraph(text="Description")))
+
+        tbody = nodes.tbody("")
+        group.append(tbody)
+
+        for key, value in pn.config.__dict__.items():
+            row = nodes.row("")
+            tbody.append(row)
+
+            default_value = value
+            description = (  # TODO: Read this from the configuration registry
+                "A (typically small) value that is per default added to the diagonal "
+                "of covariance matrices in order to make inversion numerically stable."
+            )
```

Comment:
Here we have to
- [ ] use the default value instead of the value in the documentation
- [ ] read the description from the global `Configuration` object